contracts/RevisedPixelmonTrainerAdventure/PxTrainerAdventure.sol

```diff
@@ -0,0 +1,380 @@
+        _treasure.claimedToken = 0;
+        treasures[_index] = _treasure;
+    }
+
+    /// @notice Add Sponsored Trips treasure to the smart contract
+    /// @dev Can only be called by administrators
+    /// @param _treasure Sponsored Trips information according to Treasure struct
+    function addSponsoredTripTreasure(Treasure memory _treasure) external onlyAdmin(msg.sender) {
+        if (_treasure.claimedToken != 0 || _treasure.contractType != ERC_1155_TYPE || _treasure.tokenIds.length > 0) {
+            revert InvalidInput();
+        }
+        sponsoredTrip = _treasure;
+    }
+
+    /// @notice claim function fo the winner
+    /// @dev Only winner of the week can call this method
+    /// @param _weekNumber The week number to claim treasure
+    /// @param _signature Signature from signer wallet
+    function claimTreasure(uint256 _weekNumber, bytes calldata _signature) external noContracts nonReentrant {
+        if (!(block.timestamp >= weekInfos[_weekNumber].claimStartTimeStamp && block.timestamp <= weekInfos[_weekNumber].endTimeStamp)) {
+            revert InvalidClaimingPeriod();
+        }
+        bool isValidSigner = pxChainlinkManagerContract.recoverSignerFromSignature(
+            _weekNumber,
+            weekInfos[_weekNumber].winners[msg.sender].claimed,
+            msg.sender,
+            _signature
+        );
+
+        if (!isValidSigner) {
+            revert InvalidSignature();
+        }
+
+        if (weekInfos[_weekNumber].winners[msg.sender].claimLimit == 0) {
+            revert NotAWinner();
+        }
+        if (weekInfos[_weekNumber].winners[msg.sender].claimed == weekInfos[_weekNumber].winners[msg.sender].claimLimit) {
+            revert AlreadyClaimed();
+        }
+        if (weekInfos[_weekNumber].winners[msg.sender].claimed == 0) {
+            primaryClaim(_weekNumber);
+        } else {
+            secondaryClaim(_weekNumber);
+        }
+    }
+
+    /// @notice Method to claim the first treasure
+    /// @dev This method is also used to claim Sponsor Trips if
+    ///      the caller is selected as a sponsored trip winner
+    /// @param _weekNumber The week number to claim treasure
+    function primaryClaim(uint256 _weekNumber) internal {
+        Week storage week = weekInfos[_weekNumber];
+        if (week.tripWinnersMap[msg.sender]) {
+            sponsoredTripWinners[msg.sender] = true;
+            week.tripWinnersMap[msg.sender] = false;
+
+            unchecked {
+                week.winners[msg.sender].claimed++;
+                week.availabletripsCount--;
+                sponsoredTrip.claimedToken++;
+            }
+            transferToken(_weekNumber, sponsoredTrip);
+        } else {
+            uint256 randomNumber = getRandomNumber();
+            uint256 random = randomNumber - ((randomNumber / week.remainingSupply) * week.remainingSupply) + 1;
```

Comment:
no. It won't be always 1.
It would be 1 to week.remainingSupply.

let's say,
random = X,
randomNumber = A,
week.remainingSupply = B,

X= A - ( ( A / B ) * B ) +1,

( A / B ) * B is the closest smaller number of  A  which is a multiple of B,

hence,
A - ( ( A / B ) * B ) is equal to A % B

which can be 0 to A-1.

That's why we have added 1.
Now. the result will be 1 to A.

Reply:
Here, A and B both are integer.
so, A/B will be an integer value.

so, ( A / B ) * B will not be equal to A. 
it will be the closest smaller number of A which is a multiple of B,